File: codes/java/src/main/java/com/wixigo/codeeval/PrimePalindrome.java
package com.wixigo.codeeval;

/**
 * Prime Palindrome
 * 
 * Write a program which determines the largest prime palindrome less than 1000.
 * 
 * Input sample: There is no input for this program.
 * 
 * Output sample: Print to stdout the largest prime palindrome less than 1000.
 * (soit 929)
 *
 * https://www.codeeval.com/open_challenges/3/
 * 
 * @author jcatric
 */
public class PrimePalindrome {

	public static void main(String[] args) {
		int nbr = 2;
		String palindrome = null;

		while (nbr < 1000) {
			boolean estPremier = true;
			int limit = (int) Math.sqrt(nbr);

			for (int i = 2; i <= limit; i++) {
				if (nbr % i == 0) {
					estPremier = false;
					break;
				}
			}

			if (estPremier) {
				String srtNb = String.valueOf(nbr);

				if (srtNb.length() == 1) {
					palindrome = srtNb;
				} else {
					String left = null;
					String right = null;
					int q = srtNb.length() / 2;
					if (srtNb.length() % 2 == 0) {
						left = srtNb.substring(0, q);
						right = srtNb.substring(q, srtNb.length());
					} else {
						left = srtNb.substring(0, q);
						right = srtNb.substring(q + 1, srtNb.length());
					}

					StringBuffer sbRight = new StringBuffer(right);

					if (left.equals(sbRight.reverse().toString())) {
						palindrome = srtNb;
					}
				}
			}

			nbr++;
		}

		System.out.println(palindrome);
	}

}

File: codes/js/PrimePalindrome.js

/**
 * Function to reverse a string
 *
 * @param s the string to reverse
 *
 * @returns {string} the string reversed
 */
function strrev(s) {
    var r = '';

    for (var i = s.length -1; i >=0; i--) {
        r += s[i];
    }

    return r;
}

//Le plus grand palindrome d'un nombre premier
var palindrome = '';

var nbr = 2;
while (nbr < 1000) {
    var estPremier = true;
    var limit = Math.sqrt(nbr);
    limit = Math.ceil(limit);

    for (var i = 2; i <= limit; i++) {
        if (nbr % i === 0) {
            estPremier = false;
            break;
        }
    }

    if (estPremier) {
        var strNum = '' + nbr;

        if (strNum.length === 1) {
            palindrome = strNum;
        } else {
            var left = null;
            var right = null;
            var q = (strNum.length / 2 >> 0); //Division entiere

            if (strNum.length % 2 === 0) {
                left = strNum.substring(0, q);
                right = strNum.substring(q, strNum.length);
            } else {
                left = strNum.substring(0, q);
                right = strNum.substring(q + 1, strNum.length);
            }

            if (left == strrev(right)) {
                palindrome = strNum;
            }
        }
    }

    nbr++;
}

console.log(palindrome);
